### PR TITLE
[13.0] ISPN-13463 Expose JGroups metrics

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13463

### CODE REVIEW

File: component-processor/pom.xml [hunk 1]
```diff
@@ -23,6 +23,10 @@
          <groupId>org.kohsuke.metainf-services</groupId>
          <artifactId>metainf-services</artifactId>
       </dependency>
+      <dependency>
+         <groupId>org.jgroups</groupId>
+         <artifactId>jgroups</artifactId>
```
Comment (pruivo) [inline component-processor/pom.xml:28]: optional?

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
merged! thanks @tristantarrant !

**Comment by tarioch** [comment]
With this I'm seeing now a NPE (mc is null) as we don't have a dependency to SmallRye and it doesn't look like this is a hard dependency from infinispan. Shouldn't this do nothing if you don't have a metrics collector)

```java
2022-03-17T06:52:03,262 ERROR [main] org.jgroups.JChannel           - JGRP000016: Fehler im channelConnected Callback: java.lang.NullPointerException java.lang.NullPointerException
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.lambda$channelConnected$9(JGroupsTransport.java:1512)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.channelConnected(JGroupsTransport.java:1508)
	at org.jgroups.JChannel.lambda$notifyChannelConnected$1(JChannel.java:1104)
	at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:807)
	at java.base/java.util.concurrent.CopyOnWriteArraySet.forEach(CopyOnWriteArraySet.java:425)
	at org.jgroups.JChannel.notifyListeners(JChannel.java:1118)
	at org.jgroups.JChannel.notifyChannelConnected(JChannel.java:1104)
	at org.jgroups.JChannel.connect(JChannel.java:363)
	at org.jgroups.JChannel.connect(JChannel.java:352)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.startJGroupsChannelIfNeeded(JGroupsTransport.java:548)
	at org.infinispan.remoting.transport.jgroups.JGroupsTransport.start(JGroupsTransport.java:476)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:42)
	at org.infinispan.remoting.transport.jgroups.CorePackageImpl$1.start(CorePackageImpl.java:27)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.invokeStart(BasicComponentRegistryImpl.java:617)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:608)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.access$700(BasicComponentRegistryImpl.java:30)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:635)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:599)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.access$700(BasicComponentRegistryImpl.java:30)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startDependencies(BasicComponentRegistryImpl.java:635)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.doStartWrapper(BasicComponentRegistryImpl.java:599)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.startWrapper(BasicComponentRegistryImpl.java:577)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl.access$700(BasicComponentRegistryImpl.java:30)
	at org.infinispan.factories.impl.BasicComponentRegistryImpl$ComponentWrapper.running(BasicComponentRegistryImpl.java:808)
	at org.infinispan.factories.AbstractComponentRegistry.internalStart(AbstractComponentRegistry.java:354)
	at org.infinispan.factories.AbstractComponentRegistry.start(AbstractComponentRegistry.java:250)
	at org.infinispan.manager.DefaultCacheManager.internalStart(DefaultCacheManager.java:766)
	at org.infinispan.manager.DefaultCacheManager.start(DefaultCacheManager.java:734)
	at org.infinispan.manager.DefaultCacheManager.<init>(DefaultCacheManager.java:405)
	at org.infinispan.jcache.embedded.JCacheManager.<init>(JCacheManager.java:109)
	at org.infinispan.jcache.embedded.JCachingProvider.createCacheManager(JCachingProvider.java:46)
	at org.infinispan.jcache.AbstractJCachingProvider.getCacheManager(AbstractJCachingProvider.java:66)
	at org.infinispan.jcache.AbstractJCachingProvider.getCacheManager(AbstractJCachingProvider.java:79)
```

**Comment by pruivo** [comment]
@tarioch created https://issues.redhat.com/browse/ISPN-13777
Thanks!

**Comment by pruivo** [comment]
@tarioch did it break anything in your code? I was able to reproduce the issue but the "only" effect is the exception being logged. Everything works fine.

**Comment by tarioch** [comment]
No, it's exactly the same, everything still seems to work fine.

**Pull request merged**